torch_audiomentations/augmentations/impulse_response.py

```diff
@@ -0,0 +1,32 @@
+import numpy as np
+import random
+import torch
+from torch_audiomentations.core.transforms_interface import BasicTransform
+from torch_audiomentations.utils.convolution import convolve
+from torch_audiomentations.utils.file import find_files, load_audio
+
+
+class ApplyImpulseResponse(BasicTransform):
+    """
+    Convolves an input signal with an impulse response.
+    """
+
+    def __init__(self, ir_path, convolve_mode="full", p=0.5):
+        super(ApplyImpulseResponse, self).__init__(p)
+        self.ir_path = find_files(ir_path)
+        self.convolve_mode = convolve_mode
+
+    def randomize_parameters(self, samples, sample_rate):
+        super(ApplyImpulseResponse, self).randomize_parameters(samples, sample_rate)
+
+        if self.parameters["should_apply"]:
+            self.parameters["ir_file_path"] = random.choice(self.ir_path)
+
+    def apply(self, samples, sample_rate):
+        device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
```

Comment:
I wouldn't force the device to be GPU if there is one. 
We'll probably inherit from `nn.Module` at one point (right?) and can handle device placement after that.

Reply:
I agree. I had questions myself about this so I'm glad you pointed it out. Basically, if we make all transforms inherit from PyTorch's `Module` class, then the user applying transforms can decide to place it to a `torch.device` of their choice.

This change should be pretty easy to do.